### PR TITLE
Add a automated test for testing restart

### PR DESCRIPTION
**Pull request by Yin-YinjianZhao**
In this PR, an automated test using the input based on `Examples/Physics_applications/plasma_acceleration/inputs_3d_boost` is added, in which a script is used to compare the output particle info with a corresponding one of a restart run.

### CODE REVIEW

**Comment by RemiLehe** [comment]
Thanks for this PR @Yin-YinjianZhao!
I just updated it to make the new script executable. (This was causing a failure in the CI.)

File: Regression/WarpX-tests.ini [hunk 1]
```diff
@@ -823,6 +823,25 @@ compareParticles = 0
 particleTypes = electrons
 tolerance = 1.e-14
 
+[restart]
```
Comment (ax3l) [inline Regression/WarpX-tests.ini:826]: Awesome! Can we add a 2D and RZ variant of this in the PR? :)

**Comment by RemiLehe** [comment]
Thanks for this PR @Yin-YinjianZhao !
It looks like the loops that you added to check the re-ordered particles is making the CI fail: I get the following output:
```
./analysis_restart.py  restart_plt00010
tolerance =  2.220446049250313e-16
Traceback (most recent call last):
  File "./analysis_restart.py", line 49, in <module>
    assert( num < len(zb0) )
AssertionError
```
Do you think that you could take a look at what is happening here?

Note that you can reproduce this with the following command:
```
./run_test.sh restart
```

**Comment by ax3l** [comment]
Just spotted by @RemiLehe in our weekly review:
This test just needs a new checksumAPI json file:
```
FileNotFoundError: [Errno 2] No such file or directory: '../../../../warpx/Regression/Checksum/benchmarks_json/restart.json'
```
https://warpx.readthedocs.io/en/latest/developers/checksum.html#evaluate-a-checksum-regression-test-from-a-bash-terminal

**Comment by Yin-YinjianZhao** [comment]
> /Regression/Checksum/benchmarks_json/restart.json

@ax3l 
So, I should create a file `/Regression/Checksum/benchmarks_json/restart.json`?
What content should be in this `restart.json` file?

**Comment by RemiLehe** [comment]
@Yin-YinjianZhao See [this section of the WarpX documentation](https://warpx.readthedocs.io/en/latest/developers/checksum.html#evaluate-a-checksum-regression-test-from-a-bash-terminal) (it is also the same as the one posted by Axel above): this explains how to automatically generate the `.json` file. Let me know if you have any questions.

**Review by RemiLehe — APPROVED** [review]

**Comment by RemiLehe** [comment]
Thanks a lot!

**Pull request merged**